src/runtime_src/core/tools/xbtracer/src/lib/logger.h

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <cstring>
+#include <ctime>
+#include <fstream>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+#ifdef _WIN32
+  #include <windows.h>
+#endif /* #ifdef _WIN32 */
+
+namespace xrt::tools::xbtracer {
+
+#ifdef _MSC_VER
+  // Visual Studio
+  #define FUNCTION_SIGNATURE __FUNCSIG__
+#else
+  // GCC or other compilers
+  #define FUNCTION_SIGNATURE __PRETTY_FUNCTION__ // NOLINT
+#endif /* #ifdef _MSC_VER */
+
+constexpr const char* xrt_trace_filename = "trace.txt";
+constexpr const char* xrt_trace_bin_filename = "memdump.bin";
+/*
+ * enumration to identify the log type (Entry/Exit)
+ * */
+enum class trace_type {
+  TRACE_TYPE_ENTRY,
+  TRACE_TYPE_EXIT,
+  TRACE_TYPE_INVALID
+};
```

Comment:
Do not capitalize enumerators in class enum and don't bother prefixing them.  The enumerators here should simply be: `entry`, `exit`, `invalid`.

Reply:
Sure, I have made changes and submit it in upcoming update